sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFn.java

```diff
@@ -415,6 +471,140 @@ public BoundedWindow window() {
   }
 
   /**
+   * Annotation for the method that maps an element to an initial restriction for a <a
+   * href="https://s.apache.org/splittable-do-fn">splittable</a> {@link DoFn}.
+   *
+   * <p>Signature: {@code RestrictionT getInitialRestriction(InputT element);}
```

Comment:
Is the `InputT` optional? What if the `RestrictionT` is independent of it?

Also, I like this trend of including the `Signature` section in each annotation.